lib/sorbet-rails/model_rbi_formatter.rb

```diff
@@ -215,7 +215,20 @@ def populate_collection_assoc_getter_setter(assoc_name, reflection)
 
   def populate_generated_enum_methods
     @model_class.defined_enums.each do |enum_name, enum_hash|
-      @generated_class_sigs["#{enum_name.pluralize}"] = { ret: "T::Hash[T.any(String, Symbol), Integer]"}
+      # Rails documents these values as being integers, but in practice any
+      # type is allowed.
+      value_types = enum_hash.values.map {|x| x.class.to_s }.uniq
+      value_type = if value_types.length > 1
+        "T.any(%s)" % value_types.join(', ')
+      elsif value_types.length == 1
+        # This case not strictly required, it's just neater.
+        value_types.join(', ')
+      else
```

Comment:
This would mean no `value_types` is defined. When does that happen?

Reply:
Mostly defensive coding. `enum incomplete: {}` is syntactically correct and might be part of a WIP.

I can see an argument for "raising an error" somehow but not sure the right place: I don't think this generation process should be interrupted. Could use `T.untyped`? Or maybe there's a sorbet equivalent to `T.error`?